pep-0708.rst

```diff
@@ -0,0 +1,849 @@
+protections alongside the explicit configuration, we are able to protect all
+users by default.
+
+Additionally, relying on only explicit configuration also means that every end
+user has to resolve the same problem over and over again, even in cases like
+mirrors of PyPI, Piwheels, PyTorch, etc. In each and every case they have to sit
+there and make decisions (or find some example to cargo cult) in order to be
+secure. Adding extra features into the mix allows us to centralize those
+protections where we can, while still giving advanced end users the ability to
+completely control their own destiny.
+
+
+Scopes à la npm
+---------------
+
+There's been some suggestion that scopes similar to how npm has implemented them
+may ultimately solve this. Ultimately scopes do not change anything about this
+problem. As far as I know scopes in npm are not globally unique, they're tied to
+a specific registry just like unscoped names are. However what scopes do enable
+is an obvious mechanism for grouping related projects and the ability for a user
+or organization on npm.org to claim an entire scope, which makes explicit
+configuration significantly easier to handle because you can be assured that
+there's a whole little slice of the namespace that wholly belongs to you, and
+you can easily write a rule that assigns an entire scope to a specific non
+public registry.
+
+Unfortunately, it basically ends up being an easier version of the idea to only
+use explicit configuration, which works ok in npm because its not particularly
+common for people to use their own registries, but in Python we encourage you to
+do just that.
+
+
+Open questions
+==============
+
+1. The `original proposal document <https://docs.google.com/document/d/184fQkb6NggVQfYmjTDA7p_U3iWDKk6grc2DigT1X3Es/>`__
+   was targeted more specifically to a change to pip, and went into more
+   specific details as to what we expected from pip. Since dictating UX to
+   installers isn't something that we do in PEPs, I've rewritten those parts to
+   be more generic, however that means that we lose the information on
+   repository files. Is that fine? Or should we standardize what a repository
+   file looks like so the same file can be given to multiple installers instead
+   of hand waving around the specific mechanism installers would use for
+   explicit configuration?
+2. Is the Appendix section on communicating the change useful or confusing?
+
+
+Appendix: Communicating the Change
+==================================
+
+*Note: This is pip specific and assumes specifics about how pip will choose to
+implement this PEP, it's included as an example of how we can communicate this
+change out to end users as the ecosystem rolls this change out. The entire
+Appendix should be considered to be a single communication (blog post, discuss
+post, email, whatever).*
+
+
+There's a long-standing class of attacks that are called "dependency confusion"
```

Comment:
Just to note, this would of course need some copyediting if we do end up using this as the basis to send out user-facing communications.

Reply:
Added a note to this effect.